config/webpack/production.js

```diff
@@ -27,9 +30,41 @@ module.exports = merge(sharedConfig, {
         const transformModule  = m  => m.modules ? transformModules(m.modules) : m.name;
         const transformModules = ms => ms.flatMap(m => transformModule(m));
 
-        var modules = transformModules(data.modules).sort();
+        var modules = transformModules(data.modules)
         modules = [...new Set(modules)]; // uniq
-        return JSON.stringify(modules, null, 2);
+
+        // find the package.json file for each module
+        modules = modules.filter(m => m.includes("node_modules"));
+        var packagePaths = modules.map(m => {
+          var match = m.match(/(.*node_modules\/)([^\/]+)(\/[^\/]+)?/);
+          var path = [
+            `${match[1]}${match[2]}/package.json`,
+            `${match[1]}${match[2]}${match[3]}/package.json`,
+          ].find(p => fs.existsSync(p));
+
+          if (path == null) {
+            console.warn(`[webpack-manifest] Unable to find a package.json for ${m}`);
+          }
```

Comment:
@himdel For now I put this warning here, because there's a handful of packages that don't line up.  I'm not sure what to do with them.  Here's the current list:

```
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!/Users/jfrey/.gem/ruby/2.6.5/bundler/gems/manageiq-providers-lenovo-f1c4a54ce47c/app/javascript/components/style/firmware-update-field.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./node_modules/spin.js/spin.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./app/stylesheet/application-webpack.scss
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/kubernetes-topology-graph/dist/topology-graph.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/c3/c3.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/patternfly-bootstrap-treeview/dist/bootstrap-treeview.min.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/angular-patternfly/dist/styles/angular-patternfly.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/@pf3/timeline/style.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/@manageiq/react-ui-components/dist/quadicon.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/@manageiq/react-ui-components/dist/textual_summary.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/@manageiq/react-ui-components/dist/wooden-tree.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/@manageiq/ui-components/dist/css/ui-components.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./node_modules/xml_display/XMLDisplay.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./node_modules/@manageiq/react-ui-components/dist/toolbar.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./app/javascript/tagging/sass/style.scss
[webpack-manifest] Unable to find a package.json for ./node_modules/ng-annotate-loader/loader.js!./node_modules/angular-ui-codemirror/src/ui-codemirror.js
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./node_modules/codemirror/lib/codemirror.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./node_modules/codemirror/theme/eclipse.css
[webpack-manifest] Unable to find a package.json for ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!./node_modules/@manageiq/ui-components/dist/css/ui-components.css
```

I can remove this warning if you think it's too noisy.

Reply:
Ah, those are all files that should be skipped, since there's no `node_modules` in the actual file path...

We may need to split by `!` and apply the regex only to the rightmost part.

> `./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/src??ref--7-2!./node_modules/resolve-url-loader!./node_modules/sass-loader/dist/cjs.js??ref--7-4!/Users/jfrey/.gem/ruby/2.6.5/bundler/gems/manageiq-providers-lenovo-f1c4a54ce47c/app/javascript/components/style/firmware-update-field.css`

This just means it's using `css-loader`, `postcss-loader`, `resolve-url-loader` and `sass-loader` to process `/Users/jfrey/.gem/ruby/2.6.5/bundler/gems/manageiq-providers-lenovo-f1c4a54ce47c/app/javascript/components/style/firmware-update-field.css`, which is a local file, not from a npm dependency.

Reply:
ok, I'll put in the change to split by ! and use the rightmost part.